Composer/packages/client/src/store/action/dialog.ts

```diff
@@ -1,159 +1,136 @@
-  if (!isStackEmpty) {
-    dialogs = reject(dialogs, ['id', id]);
+export const createDialogBase: ActionCreator = async (store, { id, content }) => {
+  const fixedContent = autofixReferInDialog(id, content);
+  const onCreateDialogComplete = store.getState().onCreateDialogComplete;
+  if (typeof onCreateDialogComplete === 'function') {
+    setTimeout(() => onCreateDialogComplete(id));
   }
-  return [{ id, content: dialog ? dialog.content : {}, dialogs }];
+  store.dispatch({
+    type: ActionTypes.CREATE_DIALOG,
+    payload: { id, content: fixedContent },
+  });
 };
 
-const getDiff = (dialogs1: DialogInfo[], dialogs2: DialogInfo[]) => {
-  const temp = [];
-  dialogs1.forEach(dialog => {
-    temp[dialog.id] = true;
+export const updateDialogBase: ActionCreator = async (store, { id, content }) => {
+  store.dispatch({
+    type: ActionTypes.UPDATE_DIALOG,
+    payload: { id, content },
   });
-  for (const dialog of dialogs2) {
-    if (!temp[dialog.id]) {
-      return { id: dialog.id, content: dialog.content };
-    }
-  }
 };
 
-export const removeDialogBase: ActionCreator = async (store, id) => {
-  try {
-    const projectId = store.getState().projectId;
-    const response = await httpClient.delete(`/projects/${projectId}/dialogs/${id}`);
-    luFileStatusStorage.removeFile(store.getState().botName, id);
-    store.dispatch({
-      type: ActionTypes.REMOVE_DIALOG,
-      payload: { response },
-    });
-    navTo(store, 'Main');
-  } catch (err) {
-    setError(store, {
-      message: err.response && err.response.data.message ? err.response.data.message : err,
-      summary: 'DELETE DIALOG ERROR',
-    });
-  }
+export const removeAllRelatedFiles: ActionCreator = async (store, id) => {
+  removeDialogBase(store, id);
+  //remove dialog should remove all locales lu and lg files
+  const { luFiles, lgFiles } = store.getState();
+  const luIds = luFiles.filter(file => getBaseName(file.id) === id);
+  const lgIds = lgFiles.filter(file => getBaseName(file.id) === id);
+  luIds.forEach(({ id }) => removeLuFile(store, id));
+  lgIds.forEach(({ id }) => removeLgFile(store, id));
+
+  navTo(store, 'Main');
 };
 
-export const createDialogBase: ActionCreator = async (store, { id, content }) => {
-  try {
-    const projectId = store.getState().projectId;
-    const response = await httpClient.post(`/projects/${projectId}/dialogs`, { id, content, projectId });
-    const onCreateDialogComplete = store.getState().onCreateDialogComplete;
-    if (typeof onCreateDialogComplete === 'function') {
-      setTimeout(() => onCreateDialogComplete(id));
-    }
-    store.dispatch({
-      type: ActionTypes.CREATE_DIALOG_SUCCESS,
-      payload: {
-        response,
-      },
-    });
-  } catch (err) {
-    setError(store, {
-      message: err.response && err.response.data.message ? err.response.data.message : err,
-      summary: 'CREATE DIALOG ERROR',
-    });
+export const createAllRelatedFiles: ActionCreator = async (
```

Comment:
ditto, shouldn't be part of the action file

Reply:
other 'actions' have the same problem, we need to differentiate actions (data payload) from reducer logic (bussiness logic with HTTP call or complicated transformation), otherwise the core store state will be unpredictable

(we are not far from that, just move them to another file will help)